pallets/grant/src/lib.rs

```diff
@@ -276,37 +285,36 @@ pub mod pallet {
 			// Populate Requesters struct
 			let requesters = Requesters::<T> {
 				owner: grant_receiver.clone(),
-				balance: Some(balance),
 				block_number: <frame_system::Pallet<T>>::block_number(),
 			};
 			
 			// Get hash of profile
 			let requesters_id = T::Hashing::hash_of(&requesters);
 
-			// Insert profile into HashMap
-			<StorageRequesters<T>>::insert(grant_receiver, requesters);
-
 			// Increase count for requesters
 			let new_count = Self::requesters_count().checked_add(1).ok_or(<Error<T>>::TooManyRequesters)?;
 			<RequestersCount<T>>::put(new_count);
 
+			// Insert profile into HashMap
+			<StorageRequesters<T>>::insert(grant_receiver, requesters);
+
+			
 			Ok(requesters_id)
 		}
 
 		fn select_winner() -> Result<(), DispatchError> {
 
 			let requestor: Vec<T::AccountId> = <StorageRequesters<T>>::iter_keys().collect();
 
-			// Generate randomness
-			// let _random_material = <pallet_randomness_collective_flip::Pallet<T>>::random_material();
+			//TODO: come up with a proper seed. Unix time in ns?
```

Comment:
This is important. Find out how randomness is being generated in other substrate pallets.

Reply:
https://github.com/UniversalDot/universal-dot-node/pull/107/commits/c43aaf5a431b2faef5988a19e5de050d3094a13a